pkgs/build-support/mkshell/minimal.nix

```diff
@@ -0,0 +1,47 @@
+{ writeTextFile, writeScript, bash }:
+
+# A special kind of derivation that is only meant to be consumed by the
+# nix-shell. This differs from the traditional `mkShell` in that it is
+# ultra minimalist in dependencies.
+# It does not come with traditional stdenv (i.e. coreutils).
+# It's only dependency is essentially bash.
+{
+# a list of packages to add to the shell environment
+# we simplify here rather than `native` or `propagated` since
+# this is only ever used to be in a nix-shell; which is by default
+# the current system
+packages ? [ ], ... }@attrs:
+
+derivation {
+  name = "minimal-nix-shell";
+
+  # What is the need for this reference to stdenv?
+  # Well Nix itself, checks to see if this attribute exists
+  # and sources the file.
+  # We will setup our nix-shell environment here and do things
+  # like clear the $PATH
+  # reference: https://github.com/NixOS/nix/blob/94ec9e47030c2a7280503d338f0dca7ad92811f5/src/nix-build/nix-build.cc#L494
+  "stdenv" = writeTextFile rec {
+    name = "setup";
+    executable = true;
+    destination = "/${name}";
+    text = ''
+      set -e
+      unset PATH
+
+      for a in $packages; do
+        export PATH=$a/bin:$PATH
+      done
+    '';
+  };
```

Comment:
I need to figure out how $PATH is cleared when `nix-shell --pure`.
I think it's somewhere [in this file](https://github.com/NixOS/nix/blob/94ec9e47030c2a7280503d338f0dca7ad92811f5/src/nix-build/nix-build.cc#L494) but I can't seem to trigger it.

I consider that the remaining issue to resolve before this is in a mergeable state.
(More documentation would be needed)